monitoring/v3/src/main/java/CreateCustomMetric.java

```diff
@@ -0,0 +1,324 @@
+import com.google.api.services.monitoring.v3.model.MetricDescriptor;
+import com.google.api.services.monitoring.v3.model.MonitoredResource;
+import com.google.api.services.monitoring.v3.model.Point;
+import com.google.api.services.monitoring.v3.model.TimeInterval;
+import com.google.api.services.monitoring.v3.model.TimeSeries;
+import com.google.api.services.monitoring.v3.model.TypedValue;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+
+import org.joda.time.DateTime;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.TimeZone;
+
+
+
+
+/**
+ * Class to demonstrate creating a custom metric with Cloud Monitoring. This class provides a few
+ * functions that create a custom GAUGE metric, writes a timeseries value to it, then reads that
+ * metric's value back within the last 5 minutes to see the value written.
+ */
+public class CreateCustomMetric {
+
+  /**
+   * Cloud Monitoring v3 REST client.
+   */
+  private Monitoring monitoringService;
+
+  private static SimpleDateFormat rfc3339 =
+      new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'");
+
+  static {
+    rfc3339.setTimeZone(TimeZone.getTimeZone("UTC"));
+  }
+
+  /**
+   * Identifier for project resource, in format 'projects/your-project-id'.
+   */
+  private String projectResource;
+
+  /**
+   * All custom metrics should use this domain as their prefix.
+   */
+  static final String CUSTOM_METRIC_DOMAIN = "custom.googleapis.com";
+
+  /**
+   * Name of our custom metric to create.
+   */
+  static final String DEFAULT_METRIC_TYPE = "custom_measurement";
+
+  /**
+   * The specific metric type for the instance of this class. Defaults to DEFAULT_METRIC_TYPE.
+   */
+  private String metricType;
+
+  /**
+   * The specific metric name, which is based on the project resource and the type.
+   */
+  private String metricName;
+
+
+  /**
+   * GAUGE metrics measure a value at a point in time.
+   */
+  static final String METRIC_KIND = "GAUGE";
+
+  /**
+   * Upper bound for random number to write to metric, defaults to 10.
+   */
+  private int bound = 10;
+
+  /**
+   * Constructs an instance of the class using the default metric name.
+   */
+  public CreateCustomMetric(Monitoring monitoringService, String projectResource) {
+    this.monitoringService = monitoringService;
+    this.projectResource = projectResource;
+    this.metricType = CUSTOM_METRIC_DOMAIN + "/" + DEFAULT_METRIC_TYPE;
+    this.metricName = projectResource + "/metricDescriptors/" + metricType;
+
+  }
+
+  /**
+   * Constructs an instance of the class using the default metric name, and takes in a random
+   * number generaotr (used for test purposes).
+   * Package-private to be accessible to tests.
+   */
+  CreateCustomMetric(Monitoring monitoringService, String projectResource,
```

Comment:
[nit] this line breaking is a bit weird. Most Java code I've seen in Google would break after ( and indent only 4 spaces). This horizontal alignment is allowed, but discouraged in the Java style guide. https://google.github.io/styleguide/javaguide.html#s4.6.3-horizontal-alignment